File: Javascript/7-kyu/looking-for-a-benefactor.js
function newAvg(arr, newavg) {
  if (newavg * (arr.length + 1) - arr.reduce((a, b) => a + b, 0) <= 0) {
    throw new Error('Expected New Average is too low')
  }

  return Math.ceil(newavg * (arr.length + 1) - arr.reduce((a, b) => a + b, 0))
}

File: Golang/7-kyu/looking-for-a-benefactor.go
package Golang

import "math"

func NewAvg(arr []float64, navg float64) int64 {
  var sum float64
  
  for _, dono := range arr {
    sum += dono
  }
  
  lastDonation := int64(navg * (float64(len(arr)) + 1) - math.Floor(sum))
  
  if lastDonation < 0 {
    return -1
  }
  
  return lastDonation
}
